OpenRA.Mods.Common/Orders/BasicOrderGenerator.cs

```diff
@@ -0,0 +1,18 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.Orders
+{
+	public abstract class BasicOrderGenerator
```

Comment:
What do you think about renaming this to `OrderGenerator` and having it implement `IOrderGenerator` via abstract methods. All the subclasses can then switch from `: IOrderGenerator` to `: OrderGenerator`, which would likely be cleaner.